Add docs about different formats for metadata input

LucyGwilliamAdmin opened the pull request:
Fixes #542

brockfanning commented:
@LucyGwilliamAdmin One tweak maybe: technically the normal approach is YAML *plus* Markdown. Everything above the `---` is YAML, and everything below the `---` is Markdown. It's try though that we're using the ".md" file extension. I'm not sure the best way to communicate that, but just wanted to point it out. I'll go ahead and approve this though, and leave it to you to make any changes.

LucyGwilliamAdmin commented:
@brockfanning I've just changed [line 16](https://github.com/open-sdg/open-sdg/pull/554/files#diff-1a53bdabc495e2361ad686423346c4d3R16) to try to make that clearer. I've also added a small section on naming convention, do you think that make sense?

brockfanning commented:
Looks good!
